core/src/main/scala/com/nvidia/spark/rapids/tool/Platform.scala

```diff
@@ -407,6 +414,30 @@ abstract class Platform(var gpuDevice: Option[GpuDevice],
     }
   }
 
+  private def getExecutorOffHeapMemoryMB(sparkProperties: Map[String, String]): Long = {
+    val offHeapEnabled = sparkProperties.getOrElse("spark.memory.offHeap.enabled", "false")
+      .toBoolean
+    if (offHeapEnabled) {
+      sparkProperties.get("spark.memory.offHeap.size").map {
+        size => StringUtils.convertToMB(size, Some(ByteUnit.BYTE))
+      }.getOrElse(0L)
+    } else {
+      0L
+    }
+  }
```

Comment:
For such boxed methods `get*`. The caller is in the dark whether the property was actually set in the config or not.
IF we return a `tuple<bool,value>` or `Option` then the caller can be aware that the value was not set in first place.